experimental/integrations/prototype-spark/marquez-spark-agent/src/main/java/marquez/spark/agent/lifecycle/SparkSQLExecutionContext.java

```diff
@@ -1,184 +1,187 @@
-            Collection<Path> rootPaths = JavaConversions.asJavaCollection(location.rootPaths());
-            for (Path rootPath : rootPaths) {
-              result.add(rootPath.toUri().toString());
-            }
-          } else {
-            extraInfo.append("unknown baseRelation in leaf: ").append(leaf.getClass().getName())
-            .append("\n  ").append(leaf)
-            .append("\n  ").append(lrr.getClass().getName()).append(" ").append(lrr).append("\n");
-          }
-        }else {
-          extraInfo.append("unknown leaf node: ").append(leaf.getClass().getName())
-          .append("\n  ").append(leaf).append("\n");
-        }
-    }
-    return result;
+  public void end(SparkListenerSQLExecutionEnd endEvent) {
   }
 
-  private List<String> findOutputs(LogicalPlan logical) {
-    if (logical instanceof InsertIntoHadoopFsRelationCommand) {
-      InsertIntoHadoopFsRelationCommand insert = (InsertIntoHadoopFsRelationCommand)logical;
-      return ImmutableList.of(insert.outputPath().toUri().toString());
-    } else {
-      extraInfo.append("No Output found for plan: ").append(logical.getClass().getName()).append(" ").append(logical).append("\n");
-      return ImmutableList.of();
+  @Override
+  public void setActiveJob(ActiveJob activeJob) {
+  }
+
+  @Override
+  public void start(SparkListenerJobStart jobStart) {
+    log.info("Starting job as part of spark-sql:" + jobStart.jobId());
+    QueryExecution queryExecution = SQLExecution.getQueryExecution(executionId);
+    if (queryExecution == null) {
+      log.info("No execution info {}", queryExecution);
+      return;
     }
+    LineageEvent event = LineageEvent.builder()
+        .inputs(buildInputs(queryExecution.logical()))
+        .outputs(buildOutputs(queryExecution.logical()))
+        .run(buildRun(
+            buildLogicalPlanFacet(queryExecution.logical()),
+            buildPhysicalPlanFacet(queryExecution.executedPlan())))
+        .job(buildJob())
+        .eventTime(ZonedDateTime.now())
+        .eventType("START")
+        .producer("spark")
+        .build();
+
+    marquezContext.emit(event);
   }
 
-  private void print(String prefix, SparkPlanInfo i) {
-    extraInfo.append(prefix + i.nodeName()+ "(" + i.simpleString()+ " metadata: " + i.metadata() + " metrics: "+ i.metrics() + ")" ).append("\n");
-    Collection<SparkPlanInfo> children = JavaConversions.asJavaCollection(i.children());
-    for (SparkPlanInfo child : children) {
-      print(prefix + "  ", child);
+  @Override
+  public void end(SparkListenerJobEnd jobEnd) {
+    log.info("Ending job as part of spark-sql:" + jobEnd.jobId());
+    QueryExecution queryExecution = SQLExecution.getQueryExecution(executionId);
+    if (queryExecution == null) {
+      log.info("No execution info {}", queryExecution);
+      return;
+    }
+    if (jobEnd.jobResult() instanceof JobFailed) {
+//      error = ((JobFailed)jobEnd.jobResult()).exception();
```

Comment:
not sure what facet this would be best placed at

Reply:
Ah, I had commented before seeing your comment. I thin we should add an `ErrorRunFacet` to OpenLineage
How about:
```
{
  message:
  stacktrace:
}
```